imports/redolog_manager_factory.go

```diff
@@ -0,0 +1,157 @@
+//  Copyright (c) 2017-2018 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package imports
+
+import (
+	"fmt"
+	"github.com/Shopify/sarama"
+	"github.com/uber/aresdb/common"
+	"github.com/uber/aresdb/diskstore"
+	memCom "github.com/uber/aresdb/memstore/common"
+	"github.com/uber/aresdb/metastore"
+	metaCom "github.com/uber/aresdb/metastore/common"
+	"sync"
+	"github.com/uber/aresdb/utils"
+)
+
+// Factory class to create shard level redolog manager
+type RedologManagerFactory struct {
```

Comment:
nit on the naming..
IMO usually an `XFactory` is only responsible for creating new instances of X, not keeping track of all instances created and closing them. 
this one is more like a manager of managers. (RedologDirector?)

Reply:
Yes, here you are using this struct for two purposes:
1. create an instance for RedologManager
2. holding references to some members like diskStore, metaStore

better to create difference struct for this, or for the 1st purpose, do you really need a struct to do that?